nanoFramework.Device.Bluetooth/GenericAttributeProfile/IGattServiceProvider.cs

```diff
@@ -0,0 +1,17 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+namespace nanoFramework.Device.Bluetooth.GenericAttributeProfile
+{
+    internal interface IGattServiceProvider
```

Comment:
all up question: why are those all those interfaces?
What do they bring? They're used only for 1 class usually. Do they really bring any value?
Would an abstract class be better?

Reply:
I had planned to remove those. No need for them here.
Removed